### PR TITLE
[W-10811001] Analytics ingestion on cache full

### PR DESCRIPTION
**Pull request by kchoudhari-mule**
With this PR we want to achieve 2 things:
1. Add implementation to support flush metrics when the metrics cache is full.
2. Reduce the cache size to reduce the amount of data we flush to metrics ingestion. (right now we are ingesting about 8Mb-10Mb per second, which analytics wants us to reduce to 1Mb per sec. Hence, I have reduced the cache size by 8 times).

### CODE REVIEW

File: src/main/java/com/mulesoft/runtime/services/analytics/integration/service/MetricsIngestService.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
 
 public class MetricsIngestService {
 
-    private static final long CACHE_SIZE = 16384;
+    private static final long CACHE_SIZE = 2048;
```
Comment (bahui80) [inline src/main/java/com/mulesoft/runtime/services/analytics/integration/service/MetricsIngestService.java:46]: Can't we make this a property value? This way we have a default and we can later override it quickly in main repo (OSv2) via proprty file if needed?

Comment (kchoudhari-mule) [reply src/main/java/com/mulesoft/runtime/services/analytics/integration/service/MetricsIngestService.java:46]: Discussed offline with Bahui. MQ has a different implementation for ingestion and this issue was caused by OSv2. So we are good with this change.

File: src/main/java/com/mulesoft/runtime/services/analytics/integration/service/MetricsIngestService.java [hunk 2]
```diff
@@ -75,11 +75,12 @@ public MetricsIngestService(AnalyticsClient anypointAnalyticsClient,
                                 @Value("${analytics.senderId}") String analyticsSenderId,
                                 @Value("${analytics.ingest.granularity}") int granularity,
                                 @Value("${analytics.ingest.period}") int period,
-                                @Value("${analytics.ingest.poolSize:8}") int poolSize) {
+                                @Value("${analytics.ingest.poolSize:8}") int poolSize,
+                                @Value("${analytics.ingest.cacheSize:2048}") long cacheSize) {
```
Comment (GouthamiKondakindi) [inline src/main/java/com/mulesoft/runtime/services/analytics/integration/service/MetricsIngestService.java:79]: Can we use the same property name both here and in AnalyticsAutoConfiguration

File: src/main/java/com/mulesoft/runtime/services/analytics/integration/service/MetricsIngestService.java [hunk 3]
```diff
@@ -177,6 +180,10 @@ public void updateMetricsKey(MetricsKey key, long count, long billableUnitCount,
             if (logger.isDebugEnabled()) {
                 logger.debug("metrics updated for key {}, notificationCount {}, billableUnitCount {}, byteCount {}", key, count, billableUnitCount, byteCount);
             }
+            if(metricsCache.size() >= cacheSize && getSecondsUntilNextFlush() > 0) {
```
Comment (GouthamiKondakindi) [inline src/main/java/com/mulesoft/runtime/services/analytics/integration/service/MetricsIngestService.java:183]: If we're configuring metricsCache maxSize as "cacheSize", can it's value ever exceed this cacheSize property? If not, the > check above is not necessary

**Review by bahui80 — APPROVED** [review]
LGTM

**Review by GouthamiKondakindi — APPROVED** [review]
LGTM

**Pull request merged**